### ISSUE TITLE
Windows 10 Defender flags "All Games" download as virus

### ISSUE THREAD

**Issue by capoferro** [x360ce/x360ce#1137]
This does not occur with the 64bit or 32bit downloads.

![image](https://user-images.githubusercontent.com/57349/89132868-316c4580-d4e5-11ea-9920-6013ef012c59.png)

**Comment by EJocys** [comment]
Do you have more details? Does it say more?

**Comment by EJocys** [comment]
Yeah, I can see. Reported as Trojan:Win32/Emali.A!cl - This program is dangerous and executes commands from an attacker.
I will try investigate whats caused it.

**Comment by EJocys** [comment]
Should be fine now.

I am wondering if false positive was triggered by the server side bug, which was causing X360CE fail and repeatedly upload settings to https://www.x360ce.com/webservices/x360ce.asmx?op=Execute which would INSERT, UPDATE or DELETE settings.

Today, I was scratching my head why www.x360ce.com server have so much incoming traffic.

Update:

Or it could be that execution from attacker is that unfinished "Remote Controller" feature :). Which allows for X360CE to receive controller states from another X360CE :).

![image](https://user-images.githubusercontent.com/11958537/89135067-72cf1600-d522-11ea-859c-0c0d77e10592.png)

If Defender flags X360CE again, then I will contact Microsoft. It looks like other GitHub users had this problem too.

**Comment by EJocys** [comment]
Added limit to X360CE too. Now it won't try to save settings more than 4 times. Uploaded new build:

https://github.com/x360ce/x360ce/releases

v4.12.8.0 (2020-08-02)
Fix: Server side bug was causing client fail and upload settings to cloud forever.
Fix: Bug which allowed to send settings to cloud until success (4 attempts max now).

@capoferro: Thank you for reporting.

**Comment by gandalfthewhite19890404** [comment]
And again it happened with https://github.com/x360ce/x360ce/releases/download/4.13.0.0/x360ce.zip
Windows Defender Virus Definition - 1.321..972.0 (2020-08-09)

**Comment by EJocys** [comment]
@gandalfthewhite19890404: Thank you for reporting.

I have 1.321.1004.0 version and re-downloaded x360ce.zip file. Defender passed it as safe. Maybe Microsoft fixed the issue.

https://www.microsoft.com/en-us/wdsi/definitions/antimalware-definition-release-notes

1.321.1004.0 - 2020-08-09 10:34:11 AM
1.321.972.0 - 2020-08-09 01:44:29 AM

Please [Check for Updates] of Windows Defender and try again. You should get update which is newer by 9 hours.

![image](https://user-images.githubusercontent.com/11958537/89730920-fca74380-da3a-11ea-8b01-8a1b36f8cd01.png)

Also, you can check if digital signature on your x360ce.exe file is OK i.e. original file was not altered during download?

1. You have to allow defender to keep this file.
1. Right-Click-Mouse on x360ce.exe
2. Select "Properties" menu item.
3. Check [Digital Signature] tab.

![image](https://user-images.githubusercontent.com/11958537/89731245-5872cc00-da3d-11ea-93db-bc4223589626.png)

**Comment by EvoJocys** [comment]
It is a false positive. Scanners flag the downloads by mistake from time to time; the builds are digitally signed and come back clean on VirusTotal (links in the thread).

Closing as part of a tracker clean-up after the 4.21 release.
